src/user/getUserGameList.ts

```diff
@@ -0,0 +1,79 @@
+import type {
+  AllCallOptions,
+  AuthorizationPayload,
+  UserGameListResponse
+} from "../models";
+import { call } from "../utils/call";
+import { GRAPHQL_BASE_URL } from "./USER_BASE_URL";
+
+/**
+ * This endpoint is different to others in the codebase; it hits a graphql API endpoint.
+ *
+ * The code in this file is reverse engineered from app-<hash>.js file loaded by the page
+ * at https://library.playstation.com/recently-played. Following the code in this file leads
+ * to some Apollo GraphQL code related to persisted queries. This means the request needs to
+ * contain a SHA256 hash of the GraphQL query being executed. Searching for PersistedQueryLink
+ * and createPersistedQueryLink_hashes, and an AST function in the JS source and debugging
+ * will surface the query that's passed to the hash function on the page.
+ *
+ * Thankfully it's easier to figure out future endpoints and hashes by:
+ *
+ * 1. Visiting a page, e.g https://library.playstation.com/recently-played
+ * 2. Using DevTools to find requests to https://web.np.playstation.com/api/graphql/v1/op
+ * 3. Decoding the URL parameters to find the correct SHA256 hash and some of the supported parameters
+ */
+
+type GetUserGameListOptionsCategories = "ps4_game" | "ps5_native_game";
+type GetUserGameListOptions = Pick<AllCallOptions, "limit"> & {
+  categories: GetUserGameListOptionsCategories[];
+};
+
+/**
+ * A call to this function will retrieve recently played games for the user associated
+ * with the npsso token provided to this module during initialisation.
+ *
+ * This is useful if you want recent activity that isn't tied to trophy progress.
+ *
+ * @param authorization An object containing your access token, typically retrieved with `exchangeCodeForAccessToken()`.
+ */
+export const getUserGameList = async (
```

Comment:
Is this the user's game list, recently played games, or recent activity? I am unclear because all three terms are used and can mean different things 😆

Reply:
Yeaaaah, I meant to mention this a a topic for discussion in the initial PR comment 😆 I think a name like `getRecentlyPlayed` or `getRecentlyPlayedGames` might be better, since that's what it returns. The name I gave it initially was to align it with the GraphQL operation name.